app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -712,16 +712,18 @@ class PatientRepository @Inject constructor(
     return database.patientDao().allPatientProfiles()
   }
 
-  fun saveCompleteMedicalRecord(medicalRecord: CompleteMedicalRecord) {
-    database.runInTransaction {
-      saveRecords(listOf(medicalRecord.patient))
-      if (medicalRecord.medicalHistory != null) {
-        database.medicalHistoryDao().save(medicalRecord.medicalHistory)
+  fun saveCompleteMedicalRecord(medicalRecords: List<CompleteMedicalRecord>) {
+    medicalRecords.map { medicalRecord ->
```

Comment:
This should be a `.forEach` instead of a `.map` since none of the records are being transformed in any way.